src/nemo_run/run/experiment.py

```diff
@@ -449,6 +458,30 @@ def _add_job_group(
         self._jobs.append(job_group)
         return job_group.id
 
+    def _validate_task(self, task_info: str, task: Union[Partial, Script]) -> None:
+        valid = True
+        message = ""
+        if isinstance(task, Partial):
+            serializer = ZlibJSONSerializer()
+            serialized = serializer.serialize(task)
```

Comment:
what if the serialize or deserialize fail? Shouldn't this be in a try-catch block to account for those errors as well?

Reply:
In that case it will directly raise an error with the part thats not serializable. We don't need to catch it as there's no need to reformat the error message then. The error message will look like 
```
fiddle._src.experimental.serialization.UnserializableValueError: Unserializable value <test.dummy_factory.DummyTrainer object at 0x7ac26c706510> of type <class 'test.dummy_factory.DummyTrainer'>. Error occurred at path '<root>.dummy_trainer'.")
```